packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_device.yml

```
@@ -1,1256 +1,1907 @@
+  - set:
+      field: host.name
+      copy_from: m365_defender.event.device.name
+      tag: set_host_name
+      ignore_empty_value: true
+  - lowercase:
+      field: host.name
+      tag: lowercase_host_name
+      if: ctx.host?.name != null
+  - set:
+      field: host.id
+      copy_from: m365_defender.event.device.id
+      tag: set_host_id
+      ignore_empty_value: true
+  - set:
+      field: host.ip
+      copy_from: m365_defender.event.public_ip.value
+      tag: set_host_ip
+      ignore_empty_value: true
+  - set:
+      field: host.architecture
+      copy_from: m365_defender.event.os.architecture
+      tag: set_host_architecture
+      ignore_empty_value: true
+  - set:
+      field: host.os.full
+      copy_from: m365_defender.event.os.platform
+      tag: set_host_os_full
+      ignore_empty_value: true
+  - set:
+      field: host.os.platform
+      copy_from: m365_defender.event.os.distribution
+      tag: set_host_os_platform
+      ignore_empty_value: true
+  - set:
+      field: host.os.version
+      copy_from: m365_defender.event.os.version
+      tag: set_host_os_version
+      ignore_empty_value: true
+  - set:
+      field: host.type
+      copy_from: m365_defender.event.device.type
+      tag: set_host_type
+      ignore_empty_value: true
+  - gsub:
+      field: m365_defender.event.mac_address
+      pattern: '[:.]'
+      replacement: '-'
+      tag: gsub_m365_defender_event_mac_address
+      ignore_missing: true
+  - uppercase:
+      field: m365_defender.event.mac_address
+      tag: uppercase_m365_defender_event_mac_address
+      ignore_missing: true
+  - set:
+      field: host.mac
+      copy_from: m365_defender.event.mac_address
+      tag: set_host_mac
+      ignore_empty_value: true
+
+  # Registry Mapping
+  - set:
+      field: registry.key
+      copy_from: m365_defender.event.registry.key
+      tag: set_registry_key
+      ignore_empty_value: true
+  - set:
+      field: registry.value
+      copy_from: m365_defender.event.registry.value_name
+      tag: set_registry_value
+      ignore_empty_value: true
+  - set:
+      field: registry.data.strings
+      copy_from: m365_defender.event.registry.value_data
+      tag: set_registry_data_strings
+      ignore_empty_value: true
+
+  # Source Mapping (DeviceLoginEvent has different mappings than the rest. DeviceNetworkEvent might include a direction, which controls which field goes to source/destination)
+  - set:
+      field: source.ip
+      copy_from: m365_defender.event.remote.ip
+      ignore_empty_value: true
+      tag: set_source_ip
+      if: ctx.event?.category != null && ctx.event.category.contains('authentication')
+  - set:
+      field: source.ip
+      copy_from: m365_defender.event.remote.ip
+      ignore_empty_value: true
+      tag: set_source_ip
+      if: ctx.event?.category != null && ctx.event.category.contains('network') && ctx.m365_defender?.event?.network_direction != null && ctx.m365_defender?.event?.network_direction == 'In'
```

```suggestion
      if: ctx.event?.category != null && ctx.event.category.contains('network') && ctx.m365_defender?.event?.network_direction == 'In'
```
If it is 'In' it can't be null, also below.